### PR TITLE
Feature - HomeDetail

### PR DESCRIPTION
**Pull request by at-hungpham**
Description : 
- Detail UI
- Comment API
- Play Video

### CODE REVIEW

**Comment by at-hungpham** [comment]
Screen Shot 2019-10-02 at 2 01 30 PM

File: FinalProject/View/Controller/DetailViewController/DetailViewController.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        }
+    }
+
+    func configUI() {
+        title = App.String.detailTitle
+        navigationController?.navigationBar.backgroundColor = .red
+
+        tableView.register(UINib(nibName: "DetailInfoCell", bundle: nil), forCellReuseIdentifier: "cell4")
+
+        let catagoryButton = UIBarButtonItem(image: UIImage(named: "ic-heart"), style: .plain, target: self, action: #selector(buttonDidClick))
```
Comment (at-bienle) [inline FinalProject/View/Controller/DetailViewController/DetailViewController.swift]: WARNING - Prefer object literals over image and color inits.

File: FinalProject/View/Controller/DetailViewController/DetailViewController.swift [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        authorImageView3.layer.cornerRadius = 20
+    }
+
+    @objc func buttonDidClick() {
+        print("Liked Video !")
+    }
+
+    func configThumbnail() {
+        videoImage.sd_setImage(with: URL(string: viewModel.imageURL))
+        timeLabel.text = viewModel.duration.replacingOccurrences(of: "PT", with: "").replacingOccurrences(of: "H", with: ":").replacingOccurrences(of: "M", with: ":").replacingOccurrences(of: "S", with: "")
```
Comment (at-bienle) [inline FinalProject/View/Controller/DetailViewController/DetailViewController.swift]: WARNING - Line should be 200 characters or less: currently 206 characters

**Comment by at-bienle** [comment]
Total linter violations in pull request: 2
Serious: 0
Warnings: 2

File: FinalProject/View/Controller/DetailViewController/DetailViewController.swift [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        let vc = VideoViewController()
+        vc.videoID = viewModel.id
+        present(vc, animated: true, completion: nil)
+    }
+
+    func configData() {
+        viewModel.getDataComment { [weak self] (done) in
+            guard let this = self else { return }
+            if done {
+                if this.viewModel.myComment.isNotEmpty {
```
Comment (at-bienle) [inline FinalProject/View/Controller/DetailViewController/DetailViewController.swift]: Move in another function

File: FinalProject/View/Controller/DetailViewController/DetailViewController.swift [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    @IBAction func playButton(_ sender: UIButton) {
+        let vc = VideoViewController()
+        vc.videoID = viewModel.id
+        present(vc, animated: true, completion: nil)
+    }
+
+    func configData() {
+        viewModel.getDataComment { [weak self] (done) in
```
Comment (at-bienle) [inline FinalProject/View/Controller/DetailViewController/DetailViewController.swift]: Remove ()

File: FinalProject/View/Controller/DetailViewController/DetailViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import Foundation
+import Realm
+import RealmSwift
+
+final class DetailViewModel {
+
+    var myComment: [Comment] = []
```
Comment (at-bienle) [inline FinalProject/View/Controller/DetailViewController/DetailViewModel.swift]: Rename

File: FinalProject/Model/CoreAPI/Api.Comment.swift [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import Foundation
+import UIKit
+import RealmSwift
+
+extension ApiManager.Snippet {
+
+    struct CommentResult {
+        var myComment: [Comment] = []
```
Comment (at-bienle) [inline FinalProject/Model/CoreAPI/Api.Comment.swift]: Rename

File: FinalProject/View/Controller/HomeViewController/HomeViewController.swift [hunk 7]
```diff
@@ -1,16 +1,13 @@
...
-//  FinalProject
-//
-//  Created by PCI0010 on 9/30/19.
-//  Copyright © 2019 Asiantech. All rights reserved.
-//
-
 import UIKit
 
+protocol HomeViewControllerDelegate: class {
+    func idVideo(_ homeView: HomeViewController, needforPerform videoID: Trending)
```
Comment (at-bienle) [inline FinalProject/View/Controller/HomeViewController/HomeViewController.swift]: Remove if no need

File: FinalProject/View/Controller/SearchViewController/SearchViewController.swift [hunk 8]
```diff
@@ -10,6 +10,8 @@ import UIKit
 
 final class SearchViewController: UIViewController {
 
+    @IBOutlet weak var tableView: UITableView!
```
Comment (at-bienle) [inline FinalProject/View/Controller/SearchViewController/SearchViewController.swift]: Private

File: FinalProject/View/Controller/VideoViewController/VideoViewController.swift [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+//  VideoViewController.swift
+//  FinalProject
+//
+//  Created by PCI0010 on 10/1/19.
+//  Copyright © 2019 Asiantech. All rights reserved.
+//
+import YoutubePlayer_in_WKWebView
+import UIKit
+
+class VideoViewController: UIViewController {
```
Comment (at-bienle) [inline FinalProject/View/Controller/VideoViewController/VideoViewController.swift]: final

**Comment by at-hungpham** [comment]
Screen Shot 2019-10-02 at 4 46 59 PM
Screen Shot 2019-10-02 at 4 47 16 PM

**Comment by at-bienle** [comment]
:+1: There are no linter violations.

**Pull request merged**